ledger/trie/trie.go

```diff
@@ -96,16 +96,18 @@ func (t *Trie) Insert(path ledger.Path, payload *ledger.Payload) {
 		// leaf node, which can then be populated.
 		case nil:
 
-			// When we have reached maximum depth, we can simple put a leaf node
+			// When we have reached maximum depth, we can simply put a leaf node
 			// into this location, which will then be populated in the leaf case.
 			if depth == maxDepth {
 				*current = &Leaf{}
-				return
+				continue
```

Comment:
This was a mistake I'm pretty sure since it meant that whenever we reach a nil node, we create a leaf and stop the insertion before ever computing its hash.